src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs

```diff
@@ -85,21 +97,154 @@ public static bool IsValidFormatArg(string formatString, CultureInfo formatCultu
             }
 
-            if (textFormatArgs.HasDateTimeFmt && textFormatArgs.HasNumericFmt)
+            for (int i = 0; i < formatStr.Length; i++)
             {
-                return false;
+                // Do not allow format string start with '/'
+                if (i == 0 && formatStr[i] == '/')
+                {
+                    return false;
+                }
+
+                if (_numericCharacters.Contains(formatStr[i]))
+                {
+                    // ':' is not allowed between # or 0 (numeric)
+                    if (!textFormatArgs.HasDateTimeFmt && hasColonWithNum)
+                    {
+                        return false;
+                    }
+
+                    textFormatArgs.HasNumericFmt = true;
+
+                    // Use hasNumericCharacters to check if format string has numeric character before group separator or after decimal point.
+                    hasNumericCharacters = true;
+                }
+                else if (_dateTimeCharacters.Contains(formatStr[i]))
+                {
+                    textFormatArgs.HasDateTimeFmt = true;
+                }
+                else if (!textFormatArgs.HasDateTimeFmt && formatStr[i] == ',' && !hasNumericCharacters)
+                {
+                    // If there is no numeric format character before group separator character, then treat it as an escaping character.
+                    formatStr = formatStr.Insert(i, "\\");
+                    i++;
+                }
+                else if (!textFormatArgs.HasDateTimeFmt && formatStr[i] == '.')
+                {
+                    // Reset hasNumericCharacters to false to later check if any numeric character after decimal point.
+                    decimalPointIndex = i;
+                    hasNumericCharacters = false;
+                }
+                else if (_unsupportedCharacters.Contains(formatStr[i]))
+                {
+                    // Block all unsupported characters
+                    return false;
+                }
+                else if (formatStr[i] == ':')
+                {
+                    hasColonWithNum = true;
+                }
+                else if (formatStr[i] == ';')
+                {
+                    // Does not allow number of section are more than 2.
+                    sectionCount++;
+                    if (sectionCount > 2)
+                    {
+                        return false;
+                    }
+                }
+                else if (i == formatStr.Length - 1)
+                {
+                    // If format string ends with backsplash but no following character (both numeric and datetime format), format is invalid.
+                    if (formatStr[i] == '\\')
+                    {
+                        return false;
+                    }
+
+                    // If format string ends with e or e+ (not escaping character) then format is invalid (only applied for numeric format)
```

Comment:
>only applied for numeric format [](http://example.com/codeflow?start=107&length=31)

If it is a date format I don't think we are doing the right thing. Excel treats "e" as a year specifier (`=TEXT(NOW(), "yyyy-mm-dd e")` returns "2023-08-03 2023", and if we don't handle it properly, we should block it as well (and add tests)

Reply:
Yes, block e here and add tests.